cppcache/include/geode/Region.hpp

```diff
@@ -380,22 +380,22 @@ class _GEODE_EXPORT Region : public std::enable_shared_from_this<Region> {
   template <class KEYTYPE, class VALUETYPE>
   inline void put(const KEYTYPE& key, const VALUETYPE& value,
                   const std::shared_ptr<Serializable>& arg = nullptr) {
-    put(createKey(key), createValue(value), arg);
+    put(CacheableKey::create(key), Serializable::create(value), arg);
   }
 
   /** Convenience method allowing key to be a const char* */
   template <class KEYTYPE>
   inline void put(const KEYTYPE& key, const std::shared_ptr<Cacheable>& value,
                   const std::shared_ptr<Serializable>& arg = nullptr) {
-    put(createKey(key), value, arg);
+    put(CacheableKey::create(key), value, arg);
   }
 
   /** Convenience method allowing value to be a const char* */
   template <class VALUETYPE>
   inline void put(const std::shared_ptr<CacheableKey>& key,
```

Comment:
Let's investigate this a little more. I am not excited at all by the templates that were here and tried to clean it up by removing the 4 namespaced functions that created cacheables. Because of matching rules they don't really match what you would expect. This particular function you commented on only matches if the second argument is actually of type `std::shared_ptr<Cacheable>` and will not match any of the derivatives of `Cacheable`. Almost all the calls end up matching the `KEYTYPE`, `VALUETYPE` template.